jobs/uaa/spec

```diff
@@ -546,7 +546,9 @@ properties:
         secret: app-secret
         authorized-grant-types: authorization_code,client_credentials,refresh_token
         authorities: test_resource.test_action
-        scope: test_resource.test_action,test_resource.other_action
+        scopes: # overrides anything present in 'scope' property
```

Comment:
is it wanted to change "scope" to "scopes" ?  in the diff I see at least one time still use of scope

Reply:
> _is it wanted to change "scope" to "scopes" ?_

No, we're only introducing an alternative way of defining `scope`, using an array of strings, instead of a single string of comma-separated values. So, we're expected to still support the old way of specifying scopes.

The thing is, when both `scopes` array and `scope` string are defined in UAA job properties, the latter is overwritten by the former. I'm wondering if we should merge both, instead of having one override the other. Any thoughts on this?

> _in the diff I see at least one time still use of scope_

What is implemented is a shim for transforming the `scopes` array  into the `scope` comma-separated string. In the generated `uaa.yml` file, the property name client is `scope`. That's what UAA expects in there. So in unit tests, the `all-properties-set-uaa.yml` reference only contains `scope` properties. So, that's OK to have `scope` properties still being there.

Reply:
ok